### ISSUE TITLE
WinUI-3-Demos crashes in Release mode

### ISSUE THREAD

**Issue by KiruyaMomochi** [microsoft/microsoft-ui-xaml#3400]
**Describe the bug**

WinUI-3-Demos crashes in Release mode.

**Steps to reproduce the bug**

1. Clone https://github.com/microsoft/WinUI-3-Demos
2. Open DemoBuildCs.sln in Visual Studio
3. Update WinUI Package Reference to preview 2. In DemoBuildCs.csproj:
```xml
<PackageReference Include="Microsoft.WinUI" Version="3.0.0-preview2.200713.0" />
```
4. Run app in release mode, the app start and crash immediately.

**Expected behavior**
The app runs successfully.

**Screenshots**

![image](https://user-images.githubusercontent.com/65301509/95600630-070aab80-0a85-11eb-9ba9-bbd774413674.png)

**Version Info**

NuGet package version: 
Microsoft.WinUI 3.0.0-preview2.200713.0

Windows app type:
| UWP              | Win32            |
| :--------------- | :--------------- |
|  | Yes |

| Windows 10 version                  | Saw the problem? |
| :--------------------------------- | :-------------------- |
| Insider Build (20231)              | Yes  |
| May 2020 Update (19041)            |    |
| November 2019 Update (18363)       |    |
| May 2019 Update (18362)            |    |
| October 2018 Update (17763)        |    |
| April 2018 Update (17134)          |    |
| Fall Creators Update (16299)       |    |
| Creators Update (15063)            |    |

| Device form factor | Saw the problem? |
| :----------------- | :--------------- |
| Desktop            | Yes |
| Xbox               |  |
| Surface Hub        |  |
| IoT                |  |

**Additional context**

**Comment by StephenLPeters** [comment]
Upgrading the winui3 preview package from version 1 to 2 is not supported. Unfortunately you will need to make a new project with the updated Vsix project template. This will not be the case for RTM.

**Comment by KiruyaMomochi** [comment]
Thanks for your reply.

The actual reason I open this issue is that my own app also behaves like this. 
However, my app is built from the updated project template (i.e. WinUI 3.0 Preview 2 Win32 Project).
I use the demo app to submit the issue since I think it can is easier to reproduce.

I will try the demo app with new project template then.

**Comment by KiruyaMomochi** [comment]
**Update**: I have create a new "WinUI 3.0 Preview 2 Win32" project. 
After merging the code by copying the class definition and views, it **still crash** on release mode.

```
Unhandled exception at 0x7BD972A0 (Microsoft.Internal.XamlUdk.dll) in App3.exe:
0xC000027B: An application-internal exception has occurred. (parameters: 0x1AEEF568, 0x00000003).
```

@StephenLPeters

**Comment by StephenLPeters** [comment]
@alwu-msft I'm not sure if this is in your wheelhouse or not.  @Austin-Lamb and @MikeHillberg FYI.

@KiruyaMomochi Thanks for the investigation! Can you attach the repro project to the github issue? it will make looking into this much easier.

**Comment by KiruyaMomochi** [comment]
@StephenLPeters 
Here it goes: https://github.com/KiruyaMomochi/WinUI-3-Preview2-Demo

**Comment by codendone** [comment]
I created a new project on the recently released Project Reunion 0.5 version of WinUI, and this works fine for me there.
